sweep/fee_function.go

```diff
@@ -151,6 +151,16 @@ func NewLinearFeeFunction(maxFeeRate chainfee.SatPerKWeight,
 	// Calculate how much fee rate should be increased per block.
 	end := l.endingFeeRate
 
+	// In case the starting feerate is capped at the ending feerate we make
+	// sure we have a constant fee function.
+	if start == end {
```

Comment:
I don't think this is a good move. We only allow `deltaFeeRate` to be zero when the deadline delta is 1, as otherwise there's no utility to apply a fee bumper here. It's also a question of whether we wanna create a sweeping tx given the budget is not enough, as that's why we are having this issue in the first place. By returning an error we can at least re-group this one with others, or wait until the network fee goes down.

Reply:
That's not the current behavior.

If the starting fee rate is close to the ending fee rate, the fee func can still be initialized if the width is small enough to give a non-zero fee rate delta. When the width is large, for instance, using the default 1008, then the fee rate delta may be zero due to roundings - this is why the fee rate delta is using `msat/kw` instead of `sat/kw` to increase its precision. If the starting and ending fee rates are the same, then the fee rate delta is 0, and we will still broadcast it when the width is 1.

So the question is, why would the fee rate delta be zero here - if it's due to precision, for instance, users choose a large default deadline delta, say 10000, then we should change the unit of the fee rate delta to even smaller ones; if it's due to the budget is too small like a very small budget ratio is configured, then we should increase the ratio, which is already the behavior during the fee bumping process - we'd error out when the budget is used up. Since it's almost certainly the case that, when the starting and ending fee rates are the same, because the starting fee rate is also capped at the budget fee rate, it means the estimated fee rate using the deadline as conf target must be greater than the budget, so **the budget cannot cover its fees here**.

Reply:
I think the current behaviour seems to be good as is. We don't want to publish a transaction which has no chance to be confirmed anyways. And by not sweeping this input, we increase our chances that it is regrouped with inputs with a greater budget, which can then pay for this input as well. Eventually if the deadline reaches 1, we will publish the input finally anyways

Reply:
so let's maybe increase the log information so it's easier to reason about.